stdlib/src/collections/list.mojo

```diff
@@ -604,3 +606,20 @@ struct List[T: CollectionElement](CollectionElement, Sized, Boolable):
                 result += ", "
         result += "]"
         return result
+
+
+fn count[T: ComparableCollectionElement](list: List[T], value: T) -> Int:
```

Comment:
**Suggestion** Make this a static method like we do for `__str__` [here](https://github.com/modularml/mojo/blob/d344d178aec2b80c5772261b8a6e441202a752de/stdlib/src/collections/list.mojo#L567).  This will make it easier to swap (no syntax change required on call sites of your new function) when we push `EqualityComparable` into `CollectionElement` and then can make the method non-static.  What do you think?  @gabrieldemarmiesse may able to chime in more here as he explored the different options for things like this that require trait refinement (such as your `ComparableCollectionElement`).

Reply:
The trait is fine, I'm in favor of the static method indeed as it will be easier to switch later when we have the compiler features we need. In your case the signature would look like this:

```mojo
struct List[...](...):
    ...
    @staticmethod
    fn count[U: ComparableCollectionElement](self: List[U], value: U) -> Int:
        ....


# then call it with
var my_list = List[Int](1, 2, 3, 2)
print(__type_of(my_list).count(my_list, 2))
```

Reply:
I have pushed changes for the same